pkg/azure/armauth/auth.go

```diff
@@ -76,6 +77,8 @@ func NewARMCredential(opt *Options) (azcore.TokenCredential, error) {
 		return azidentity.NewEnvironmentCredential(nil)
 	case ManagedIdentityAuth:
 		return azidentity.NewManagedIdentityCredential(nil)
+	case WorkloadIdentityAuth:
+		return azidentity.NewDefaultAzureCredential(nil)
```

Comment:
I don't think we actually need this code. It's only called for testing afaik. but in the case we ever need it, it's the right code to have

Reply:
Would a link to this doc help clarifying how this maps workload identity https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview?tabs=dotnet#azure-identity-client-libraries?

Reply:
Would be good to double-check when this code gets used... I think this might only be called when running Radius locally in the debugger. 

It might be good to think about whether we can remove it ...